### PR TITLE
Fix issue when generating the inline array to be send to the API

### PR DESCRIPTION
**Pull request by zot24**
Could you please guys @moltin/moltin have a look to my [recursion](https://github.com/moltin/php-sdk/compare/fix/issue-when-generating-inline-arrays) to see if this have sense to you and I'm not making a mistake again, this should process any case where we are generating a `inline array` as for example reordering categories of creating/updating a product (when selecting the categories)

### CODE REVIEW

**Comment by anthonysterling** [comment]
You could try extracting the logic as is, writing a test or two, then implement this @zot24. The tests should give you some feedback as to whether the behaviour changed.

**Comment by zot24** [comment]
Thanks @antonysterling I'll work on those test btw #32 will help

**Comment by zot24** [comment]
:eyes: @anthonysterling let me know your thought :)

**Comment by anthonysterling** [comment]
FYI, it looks like your target branch is the `version1` which we currently test against PHP 5.3, and you're using the short array syntax for 5.4 in your tests. ;)

**Comment by zot24** [comment]
Thanks! @anthonysterling if you got a chance :eyes: I fixed most of the staff and now the test looks much cleaner :)

File: tests/Request/CURLTest.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        );
+    }
+
+    /**
+     * Get a protected method through reflection
+     *
+     * @param $name
+     * @return \ReflectionMethod
+     */
+    protected static function getMethod($name) {
```
Comment (anthonysterling) [inline tests/Request/CURLTest.php]: There's no need to make this static in this context.

**Comment by anthonysterling** [comment]
Looking good @zot24 :+1:

**Pull request merged**